hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/spark/sql/hudi/dml/TestMergeIntoTable2.scala

```diff
@@ -1099,7 +1099,7 @@ class TestMergeIntoTable2 extends HoodieSparkSqlTestBase {
             }
 
             // Test 2: At least one partial insert assignment clause misses primary key.
-            checkException(
+            checkExceptionContain(
```

Comment:
What's reason of changing `checkException` to `checkExceptionContain`?

Reply:
So this is because of throwing a new exception `MergeIntoFieldResolutionException` by appending the error message.  Could you instead change the expected error message so that we still use `checkException` for assertion?